add favicons to webpage

ben-schwen opened the pull request:
Towards #3956

jangorecki commented:
I wonder if that could possibly live in `.graphics` directory? then we don't have extra new directory for exactly the same purpose.

jangorecki commented:
@ben-schwen yes, copy in CI feels easiest. Then we don't need to change pkgdown yaml, righr? Look at the _integration_ job. It publishes to a directory named pkgdown already, so either copy it to pkgdown/favicon (if pkgdown needs them for rendering) and then remove that directory. Or you just copy it after to pkgdown/pkgdown/favicon, if it is not needed during render website.

ben-schwen commented:
@jangorecki I adjusted the CI accordingly, now copying the files from `.graphics`. Previously we were building the pkgdown site into `./pkgdown` and this needs the folder to be empty (since we overwrite it). Therefore I changed the destination of the pkgdown build to `./website` (also in the subsequent places).

jangorecki commented:
Changing name to website looks good but I think you forget to copy icons from .graphics dir in CI, no?

ben-schwen commented:
It's done here but the line might be a bit bloated

https://github.com/Rdatatable/data.table/blob/ee600f0c23c2e353e9f41ca38f938a25caa9bb8d/.gitlab-ci.yml#L300

jangorecki commented:
Uh, yes, please put in own line, so each workaround has own line then

jangorecki commented on .gitlab-ci.yml:
```diff
@@ -297,8 +297,9 @@ integration:
   script:
     - R --version
     - *install-deps ## markdown pkg not present in r-pkgdown image
+    - rm -rf ./pkgdown && mkdir -p ./pkgdown/favicon/ && cp .graphics/favicon/* ./pkgdown/favicon/ ## copy favicons
```
```suggestion
    - mkdir -p ./pkgdown/favicon/ && cp .graphics/favicon/* ./pkgdown/favicon/ ## copy favicons
```
